apps/desktop-wallet/src/storage/tanstackQueryCache/useClearQueryCacheOnWalletLock.ts

```diff
@@ -0,0 +1,17 @@
+import { usePersistQueryClientContext } from '@alephium/shared-react'
+import { useEffect } from 'react'
+
+import { useAppSelector } from '@/hooks/redux'
+
+const useClearQueryCacheOnWalletLock = () => {
+  const walletId = useAppSelector((s) => s.activeWallet.id)
+  const { clearQueryCache } = usePersistQueryClientContext()
+
+  useEffect(() => {
+    if (!walletId) {
+      clearQueryCache()
+    }
+  }, [clearQueryCache, walletId])
+}
```

Comment:
This hook replaces all `clearQueryCache` calls that were removed above. It ensures that the cache is cleared only after the Redux dispatched events have changed the React state.